src/Linux/curl_easy.cpp

```diff
@@ -106,6 +110,14 @@ std::unique_ptr<curl_easy> curl_easy::create(const std::string& url, const std::
     easy->set_opt_or_throw(CURLOPT_FAILONERROR, 1L);
     easy->set_opt_or_throw(CURLOPT_SSLVERSION, CURL_SSLVERSION_TLSv1_2);
 
+    // If request is routed to THIM agent for SGX certificate fetch, set
+    // response
+    // timeout to 500 ms
+    if (fetchingFromLocalAgent)
+    {
+        easy->set_opt_or_throw(CURLOPT_SERVER_RESPONSE_TIMEOUT, 0.5L);

```

Comment:
`0.5L` is of type `long double` which when passed to `curl_easy_setopt` will just get truncated down to 0 because [CURLOPT_SERVER_RESPONSE_TIMEOUT](https://curl.se/libcurl/c/CURLOPT_SERVER_RESPONSE_TIMEOUT.html) expects a `long` while typecasting `double long` to `long` truncates. I think the least timeout you can set using CURLOPT_SERVER_RESPONSE_TIMEOUT is 1s. 




Alternatively, you can use `CURLOPT_TIMEOUT_MS` but this will apply to the whole request as opposed to just the server response. Personally, would recommend setting CURLOPT_SERVER_RESPONSE_TIMEOUT `1s` unless you can guarantee that client will be able to make request within less than `500ms`.

Reply:
Updated to 1 sec. Thank you for catching this. I did not think about implicit typecasting.